### PR TITLE
RSDK-7746: Add SetRPM to go SDK

### PR DESCRIPTION
**Pull request by martha-johnston**
- add `SetRPM` to the motor interface
- add `SetRPM` to the client and server files for go SDK
- add unimplemented SetRPM to all motor drivers
- add `SetRPM` to inject motor

### CODE REVIEW

File: components/motor/errors.go [hunk 1]
```diff
@@ -30,3 +30,8 @@ func NewGoToUnsupportedError(motorName string) error {
 func NewControlParametersUnimplementedError() error {
 	return errors.New("control parameters must be configured to setup a motor with controls")
 }
+
+// NewSetRPMUnsupportedError returns an error when a motor is required to support SetRPM.
```
Comment (oliviamiller) [inline components/motor/errors.go:34]: comment seems incorrect?

File: components/motor/motor.go [hunk 2]
```diff
@@ -97,8 +97,8 @@ type Motor interface {
 	// revolutions at a given speed in revolutions per minute. Both the RPM and the revolutions
 	// can be assigned negative values to move in a backwards direction. Note: if both are
 	// negative the motor will spin in the forward direction.
-	// If revolutions is 0, this will run the motor at rpm indefinitely
 	// If revolutions != 0, this will block until the number of revolutions has been completed or another operation comes in.
+	// Deprecated: If revolutions is 0, this will run the motor at rpm indefinitely.
```
Comment (oliviamiller) [inline components/motor/motor.go:101]: just curious are we going to get rid of this behavior entirely once setRPM is done?

**Review by oliviamiller — APPROVED** [review]

File: components/motor/errors.go (hunk 1 above)
Comment (martha-johnston) [reply components/motor/errors.go:34]: yeah I kind of agree but was trying to follow the pattern of the other comments/errors of this type

Comment (martha-johnston) [reply components/motor/errors.go:34]: I'll change it tho

Comment (martha-johnston) [inline components/motor/errors.go:34]:
```suggestion
// NewSetRPMUnsupportedError returns an error when a motor does not support SetRPM.
```

**Pull request merged**